src/systemctl/systemctl-list-units.c

```diff
@@ -403,19 +402,18 @@ static int output_sockets_list(struct socket_info *socket_infos, unsigned cs) {
                                 return table_log_add_error(r);
 
                 }
-
-                on = ansi_highlight();
-                off = ansi_normal();
-        } else {
-                on = ansi_highlight_red();
-                off = ansi_normal();
         }
 
         r = output_table(table);
         if (r < 0)
                 return r;
 
         if (arg_legend != 0) {
+                const char *on, *off;
+
+                on = cs > 0 ? ansi_highlight() : ansi_highlight_red();
+                off = ansi_normal();
```

Comment:
These are used only once. I think it is ok to inline them in the `printf()` below, if you'd like.

Reply:
Okay, let's try that.